File: app/src/main/java/com/thanksmister/bitcoin/localtrader/network/api/model/Advertisements.java
/*
 * Copyright (c) 2018 ThanksMister LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.thanksmister.bitcoin.localtrader.network.api.model;

import android.support.annotation.NonNull;


import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public class Advertisements {
    @SerializedName("ad_list")
    @Expose
    public List<Advertisement> items = new ArrayList<Advertisement>();

    @SerializedName("ad_count")
    @Expose
    public Integer count;

    public Advertisements(@NonNull final List<Advertisement> items) {
        this.items = items;
    }

    public List<Advertisement> getItems() {
        Collections.sort(items, new Comparator<Advertisement>() {
            @Override
            public int compare(Advertisement a1, Advertisement a2) {
                boolean b1 = a1.getVisible();
                boolean b2 = a2.getVisible();
                if (b1 && !b2) {
                    return +1;
                }
                if (!b1 && b2) {
                    return -1;
                }
                return 0;
            }
        });

        return items;
    }

    public int getCount() {
        return count;
    }
}

File: app/src/main/java/com/thanksmister/bitcoin/localtrader/network/api/model/Advertisements.kt
/*
 * Copyright (c) 2020 ThanksMister LLC
 *  http://www.thanksmister.com
 *   
 *  Mozilla Public License 2.0
 *  
 *  Permissions of this weak copyleft license are conditioned on making
 *  available source code of licensed files and modifications of those files
 *  under the same license (or in certain cases, one of the GNU licenses).
 *  Copyright and license notices must be preserved. Contributors provide
 *  an express grant of patent rights. However, a larger work using the
 *  licensed work may be distributed under different terms and without source
 *  code for files added in the larger work.
 */

package com.thanksmister.bitcoin.localtrader.network.api.model


import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.util.ArrayList
import java.util.Collections
import java.util.Comparator

class Advertisements(items: List<Advertisement>) {

    @SerializedName("ad_list")
    @Expose
    var items: List<Advertisement> = ArrayList()

    @SerializedName("ad_count")
    @Expose
    var count: Int? = null

    init {
        this.items = items
    }

    fun getListItems(): List<Advertisement> {
        Collections.sort(items, Comparator { a1, a2 ->
            val b1 = a1.visible
            val b2 = a2.visible
            if (b1 && !b2) {
                return@Comparator +1
            }
            if (!b1 && b2) {
                -1
            } else 0
        })

        return items
    }

    fun getCount(): Int {
        return count!!
    }
}
